db/migrate/20180713194201_create_canister.rb

```diff
@@ -0,0 +1,21 @@
+class CreateCanister < ActiveRecord::Migration[5.0]
+  def change
+    create_table :canisters do |t|
+      t.string :serial_number
```

Comment:
2 things here, can you add ems_ref for a unique reference?  Even if e.g. position is unique for lenovo we can use ems_ref generally.
Also I assume this belongs to a physical_storage but you don't have a reference to physical_storage here.

Reply:
@agrare I added now `ems_ref` and `physical_storage_id` to the migration.